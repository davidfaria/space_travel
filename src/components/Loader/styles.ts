import styled, { keyframes, css } from 'styled-components'

const loading = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`

interface WrapperProps {
  align: string
}
interface SpinnerProps {
  size: number
  color: string
}

export const Wrapper = styled.div<WrapperProps>`
  /* min-height: 600px;
  height: auto; */
  display: flex;
  align-items: center;
  ${({ align }) =>
    css`
      justify-content: ${align};
    `}

  span {
    font-size: 1.8rem;
    margin-left: 16px;
  }
`
export const Spinner = styled.div<SpinnerProps>`
  width: 40px;
  height: 40px;
  border: 5px solid;

  ${({ size }) => css`
    width: ${size}px;
    height: ${size}px;
  `}

  ${({ color }) =>
    color !== ''
      ? css`
          border-color: ${color} var(--border-loading) var(--border-loading);
        `
      : css`
          border-color: var(--border-loading-active) var(--border-loading)
            var(--border-loading);
        `}

  border-radius: 50%;
  animation: ${loading} 0.7s linear infinite;
`
