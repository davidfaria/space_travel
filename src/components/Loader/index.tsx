import React from 'react'

import * as S from './styles'

interface LoaderProps {
  loading?: boolean
  size?: number
  color?: string
  align?: 'left' | 'center' | 'right'
  message?: string
}

const Loader = ({
  loading = false,
  size = 60,
  color = '',
  align = 'left',
  message = ''
}: LoaderProps) => {
  if (!loading) return null
  return (
    <S.Wrapper align={align}>
      <S.Spinner data-testid="loader" size={size} color={color} />
      <span>{message}</span>
    </S.Wrapper>
  )
}

export default Loader
