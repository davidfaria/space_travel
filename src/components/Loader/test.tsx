import { render, screen } from '@testing-library/react'

import Loader from '.'

describe('<Loader />', () => {
  it('should render loader', () => {
    const { container } = render(<Loader loading={true} />)
    const spinner = screen.getByTestId('loader')
    expect(spinner).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should not render loader', () => {
    const { container } = render(<Loader />)
    expect(screen.queryByTestId('loader')).not.toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
