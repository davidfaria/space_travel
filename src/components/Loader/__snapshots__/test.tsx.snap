// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should not render loader 1`] = `null`

exports[`<Loader /> should render loader 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
}

.c0 span {
  font-size: 1.8rem;
  margin-left: 16px;
}

.c1 {
  width: 40px;
  height: 40px;
  border: 5px solid;
  width: 60px;
  height: 60px;
  border-color: var(--border-loading-active) var(--border-loading) var(--border-loading);
  border-radius: 50%;
  -webkit-animation: eoUyJr 0.7s linear infinite;
  animation: eoUyJr 0.7s linear infinite;
}

<div
  class="c0"
>
  <div
    class="c1"
    color=""
    data-testid="loader"
    size="60"
  />
  <span />
</div>
`
